ui/src/shared/components/AutoRefresh.js

```diff
@@ -150,12 +151,12 @@ const AutoRefresh = ComposedComponent => {
       Promise.all(timeSeriesPromises).then(timeSeries => {
         const newSeries = timeSeries.map(response => ({response}))
         const lastQuerySuccessful = !this._noResultsForQuery(newSeries)
-
         this.setState({
           timeSeries: newSeries,
           lastQuerySuccessful,
           isFetching: false,
         })
+        grabDataForDownload(timeSeries)
```

Comment:
What happens if `grabDataForDownload` is `undefined`?    Lots of other components use `AutoRefresh` and they wont be passing in this function.  This is might be a good use for `getDefaultProps` with `grabDataForDownload` being a noop.